dpd/src/link.rs

```diff
@@ -1742,8 +1743,18 @@ async fn reconcile_link(
             .await
             .as_qsfp()
             .cloned();
-        if let Some(qsfp) = qsfp {
-            qsfp_xcvr_mpn(&qsfp).unwrap_or(None)
+
+        if let Some(qsfp) = &qsfp {
+            match qsfp_xcvr_mpn(qsfp) {
+                Ok(mpn) => Some(mpn),
+                Err(e) => {
+                    warn!(
+                        log,
+                        "failed to get MPN for qsfp: {e}, port: {port_id}"
```

Comment:
I think we may want to use the key-value formatting in these log messages, something like:

```rust
warn!(log, "failed to get MPN for QSFP"; "port" => %port_id, "error" => %e);
```

Keeps them as distinct keys in the JSON output.